lib/plugins/aws/invokeLocal/index.js

```diff
@@ -147,10 +166,187 @@ class AwsInvokeLocal {
+          reject(new ServerlessError(`Attribute definition unexpected ${JSON.stringify(value)}`));
+        }
+
+        // Get the resource name
+        resourceName = value[action][0];
+      } else if (value['Fn::ImportValue']) {
+        action = 'Fn::ImportValue';
+        resourceName = value[action];
+      } else {
+        return resolve(value);
+      }
+
+      if (cache[action] && cache[action][resourceName]) {
+        switch (action) {
+          case 'Ref':
+          case 'Fn::ImportValue':
+            return resolve(cache[action][resourceName]);
+          case 'Fn::GetAtt':
+            if (cache[action][resourceName][value[action][1]]) {
+              return resolve(cache[action][resourceName][value[action][1]]);
+            }
+            break;
+          default:
+            return resolve(value);
+        }
+      }
+
+      if (!cache[action]) {
+        cache[action] = {};
+      }
+
+      if (action === 'Fn::ImportValue') {
+        const cf = new aws.CloudFormation(serviceConfig);
+        resolve(
+          this.loadExports(cf).then(cfExports => {
+            cfExports.forEach(exp => {
+              cache[action][exp.Name] = exp.Value;
+            });
+            return cache[action][resourceName];
+          })
+        );
+      } else {
+        let resource = this.serverless.service.resources.Resources[resourceName];
+        if (!resource) {
+          // Find the function
+          let lambdaName = resourceName;
+          let func = _.entries(this.serverless.service.functions).find(item => {
+            const [, val] = item;
+
+            return val.name === resourceName;
+          });
 
-    _.merge(process.env, lambdaDefaultEnvVars, credentialEnvVars, configuredEnvVars);
+          if (!func) {
+            func = _.entries(this.serverless.service.functions).find(item => {
+              let name = '';
+              const [functionName, itemValue] = item;
 
-    return BbPromise.resolve();
+              const lowerName =
+                functionName.substr(0, 1).toUpperCase() +
+                functionName.substr(1, functionName.length - 1);
+              name = `${lowerName}LambdaFunction`;
+
+              const resultVal = name === resourceName;
+              if (resultVal) {
+                lambdaName = itemValue.name;
+              }
+              return resultVal;
+            });
+          }
+
+          if (!func) {
+            // If no function return what we recieved
+            return resolve(value);
+          }
+
+          // Construct fake resource definition for lambda
+          resource = {
+            Type: 'AWS::Lambda::Function',
+            Properties: {
+              FunctionName: lambdaName,
+            },
+          };
+        }
+        const handler = handlers[resource.Type];
+        if (handler && handler[action]) {
+          resolve(
+            handler[action](this, resource, resourceName, value[action], serviceConfig).then(
```

Comment:
I think signature should rather be: `this, resource, resourceName`, as if I see correctly:
- `value[action]` in most cases matches `resourceName`, and is used in resolvers purely as an error fallback. Where I believe we should rather throw error instead of returning input value, as that's not what should land as env var value.
- `serviceConfig` can be resolved from `this`, and in resolvers it seems to be used for AWS SDK requests, while all such requests should be configured to go through `provider.request`

Reply:
the value[action] is used in Fn::GetAtt to be able to resolve underlying resources  An example is the sqs.js 'Fn::GetAtt' function, which uses it to identify the type of value which is needed ("Arn", "QueueName", etc)

Reply:
Looks like the first time provider.request gets used, all regular aws-sdk calls start failing.  I was able to remove all the aws-sdk direct references